goicicle/curves/bn254/bn254_test.go

```diff
@@ -0,0 +1,221 @@
+package bn254
+
+import (
+	"encoding/binary"
+	"testing"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/consensys/gnark-crypto/ecc/bn254/fp"
+)
+
+func TestNewFieldBN254One(t *testing.T) {
+	oneField := NewFieldBN254One()
+	rawOneField := [8]uint32([8]uint32{0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0})
+
+	assert.Equal(t, oneField.limbs(), rawOneField)
+}
+
+func TestNewFieldBN254Zero(t *testing.T) {
+	zeroField := NewFieldBN254Zero()
+	rawZeroField := [8]uint32([8]uint32{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0})
+
+	assert.Equal(t, zeroField.limbs(), rawZeroField)
+}
+
+func TestFieldBN254FromGnark(t *testing.T) {
+	var rand fr.Element
+	rand.SetRandom()
+
+	s := FieldBN254FromGnark(rand)
+
+	assert.Equal(t, s.limbs(), ConvertUint64ArrToUint32Arr(rand))
+}
+
+func TestFieldBN254GetLimbs(t *testing.T) {
+	zeroField := NewFieldBN254Zero()
+
+	var field FieldBN254
+
+	assert.Equal(t, zeroField.limbs(), field.limbs())
+}
+
+func TestFieldBN254ToBytesLe(t *testing.T) {
+	var rand fr.Element
+	rand.SetRandom()
+
+	f := FieldBN254FromGnark(rand)
+
+	expected := make([]byte, len(f.s)*4) // each uint32 takes 4 bytes
+	for i, v := range f.s {
+		binary.LittleEndian.PutUint32(expected[i*4:], v)
+	}
+
+	assert.Equal(t, f.toBytesLe(), expected)
+	assert.Equal(t, len(f.toBytesLe()), 32)
+}
+
+func TestNewPointBN254Zero(t *testing.T) {
+	point := NewPointBN254Zero()
+
+	assert.Equal(t, point.x, *NewFieldBN254Zero())
+	assert.Equal(t, point.y, *NewFieldBN254One())
+	assert.Equal(t, point.z, *NewFieldBN254Zero())
+}
+
+func TestBN254Eq(t *testing.T) {
+	p1 := NewPointBN254Zero()
+	p2 := NewPointBN254Zero()
+	p3 := &PointBN254{
+		x: *NewFieldBN254One(),
+		y: *NewFieldBN254One(),
+		z: *NewFieldBN254One(),
+	}
+
+	assert.Equal(t, p1.eq(p2), true)
+	assert.Equal(t, p1.eq(p3), false)
+}
+
+func TestBN254StripZ(t *testing.T) {
+	p1 := NewPointBN254Zero()
+	p2ZLess := p1.strip_z()
+
+	assert.IsType(t, PointAffineNoInfinityBN254{}, *p2ZLess)
+	assert.Equal(t, p1.x, p2ZLess.x)
+	assert.Equal(t, p1.y, p2ZLess.y)
+}
+
+func TestPointBN254FromGnark(t *testing.T) {
```

Comment:
This test is failing. It seems like its due to the implementation and not the test itself but I haven't been able to figure out why.